server/routes/v1/user.ts

```diff
@@ -1,59 +1,140 @@
-      res.status(400).send({ code: 400, message: "Missing token." });
+export async function signInHandler(req: Request, res: Response) {
+  try {
+    const { userName, password } = req.body;
+
+    if (!userName || !password) {
+      res.status(400).json({ message: "Username or password is undefined" });
       return;
     }
 
-    const status = await uContr.confirmEmail(token);
+    response = await uContr.signIn(userName, password);
+    user = response.data.result;
+
+    if (user) {
+      if (!process.env.JWT_SECRET) {
+        throw "Undefined secret";
+      }
+
+      const token = jwt.sign(
+        { username: user.userName, id: user.id },
+        process.env.JWT_SECRET,
+        { expiresIn: "1h" }
+      );
 
-    if (status) {
-      res.status(204);
-    } else {
       res
-        .status(400)
-        .send({ code: 400, message: "Token is invalid or expired." });
+        .cookie("token", JSON.stringify(token), {
+          httpOnly: true,
+          expires: moment().add(1, "hour").toDate(),
+          signed: true,
+        })
+        .status(response.status)
+        .json(response.data);
+    } else {
+      res.status(response.status).json(response.data);
     }
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "Something went wrong" });
   }
-);
+}
 
-userRouter.get(
-  "/reset-password/r=:token",
-  async (req: Request, res: Response) => {
-    if (!req.params.token || !req.body.pass || !req.body.confpw) {
-      res.status(400).send({
-        code: 400,
-        message: "Missing token or password.",
-      });
-      return;
-    }
+export async function signUpHandler(req: Request, res: Response) {
+  const { email, userName, password, firstName, lastName } = req.body;
 
-    const status = await uContr.resetPassword(
-      req.params.token,
-      req.body.pass,
-      req.body.confpw
-    );
+  if (!email || !userName || !password || !firstName || !lastName) {
+    res.status(400).json({ message: "Missing values in request body" });
+  }
 
-    if (status) {
-      res.status(204);
-    } else {
-      res
-        .status(400)
-        .send({ code: 400, message: "Token is invalid or expired." });
-    }
+  response = await uContr.createUser(
+    email,
+    userName,
+    password,
+    firstName,
+    lastName
+  );
+
+  res.status(response.status).json(response.data);
+}
+
+async function confirmEmailHandler(req: Request, res: Response) {
+  const token = req.params.token;
+
+  if (!token) {
+    res.status(400).json({ code: 400, message: "Missing token." });
+    return;
   }
-);
+
+  const status = await uContr.confirmEmail(token);
+
+  if (status) {
+    res.status(200).json({ message: "Email successfully confirmed" });
+  } else {
+    res
+      .status(400)
+      .json({ code: 400, message: "Token is invalid or expired." });
+  }
+}
+
+async function resetPassHandler(req: Request, res: Response) {
+  if (!req.params.token || !req.body.pass || !req.body.confpw) {
```

Comment:
Lets standardize on the naming convention. You use `password` in `signUpHandler` but `pass` here.

Reply:
pouya!!!

Reply:
:(